src/ble_manager.cc

```diff
@@ -42,19 +42,27 @@ template <typename O, typename M, class... Types> auto bind2(O* object, M method
     }                                                \
     BluetoothLEDevice& _device = *peripheral.device;
 
-#define CHECK_RESULT(_result)                            \
-    if (!_result)                                        \
-    {                                                    \
-        LOGE("result is null");                          \
-        return;                                          \
-    }                                                    \
-    auto _commStatus = _result.Status();                 \
-    if (_commStatus != GattCommunicationStatus::Success) \
-    {                                                    \
-        LOGE("communication status: %d", _commStatus);   \
-        return;                                          \
+template <class T>
+bool CHECK_RESULT(T _result)
```

Comment:
Maybe rename this to `CheckResult` to make it clear that this is a function

Reply:
Resolved